Framework/include/QualityControl/runnerUtils.h

```diff
@@ -121,11 +122,37 @@ inline int computeRunNumber(const framework::ServiceRegistry& services, int defa
                            "   using the one from the config file or 0 as a last resort."
                         << ENDM;
   }
-  run = run > 0 /* found it in service */ ? run : defaultRunNumber;
+  run = run > 0 /* found it in service */ ? run : fallbackRunNumber;
   ILOG(Debug, Devel) << "Run number returned by computeRunNumber (default) : " << run << ENDM;
   return run;
 }
 
+inline std::string computePeriodName(const framework::ServiceRegistry& services, const std::string& fallbackPeriodName = "")
+{ // Determine period
+  std::string periodName;
+  periodName = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("periodName", "unspecified");
+  ILOG(Info, Devel) << "Got this property periodName from RawDeviceService: '" << periodName << "'" << ENDM;
+  periodName = periodName != "unspecified" /* found it in service */ ? periodName : fallbackPeriodName;
+  ILOG(Debug, Devel) << "Period Name returned by computePeriodName : " << periodName << ENDM;
+  return periodName;
+}
+
+inline std::string computePassName(const std::string& fallbackPassName = "")
+{
+  std::string passName;
+  passName = fallbackPassName;
+  ILOG(Debug, Devel) << "Period Name returned by computePassName : " << passName << ENDM;
+  return passName;
+}
+
+inline std::string computeProvenance(const std::string& fallbackProvenance = "")
```

Comment:
I think that there is a rebase that is missing because these lines were added in another PR.

Reply:
i will check

Reply:
I do not understand what is the problem. For now there are two copies of `compute*` functions until we stop using the ones which require the whole ptree (aggregators still use them).

Reply:
oh, sorry, I had misunderstood what was going on. All good !